File: projects/yuki/yes_no.py
#!/usr/bin/env python3

def yes_no( str ):

    while (1):
        answer = input(str)
        answer = answer.lower()
        if (answer == "y" or answer == "yes" or answer == "yep"):
            return True
        elif (answer == "n" or answer == "no" or answer == "nah"):
            return False
        else:
            print("Not a valid response. Try again.")
        

if (yes_no("Does this work? ")):
    print("YES!!")
else:
    print("NO!!")

File: projects/yuki/yes_no.pl
#!/usr/bin/env perl

sub yes_no {
    my ($question) = @_;

    while (1) {

	print $question;
	my $answer = <STDIN>;
	chomp $answer;
	$answer = lc $answer;

	if ( $answer eq 'y' || $answer eq "yep" || $answer eq "yes" || $answer eq "ya" )  {
	    return true;
	} elsif ( $answer eq "n" || $answer eq "no" || $answer eq "nope" || $answer eq "nah") {
	    return false;
	} else {
	    print "Not a valid input. Try again.\n";
	}
    }
}

if ( yes_no("Does this work? ") ) {
    print("Yes\n");
} else {
    print("No\n");
}
